homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name, job_title
+FROM employees
+    INNER JOIN jobs USING (job_id)
+WHERE salary < (
+    SELECT MIN(salary)
+    FROM employees
+    WHERE job_id = 'MK_MAN'
+);
+
 --28. the employee number, name( first name and last name ) and job title for all employees whose salary is smaller than any salary of those employees whose job title is MK_MAN. Exclude Job title MK_MAN.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name, job_title
+FROM employees
+    INNER JOIN jobs USING (job_id)
+WHERE salary < (
+    SELECT MIN(salary)
+    FROM employees
+    WHERE job_id = 'MK_MAN'
+) AND job_id != 'MK_MAN'; --Redundant: at this point any salary is smaller than MIN(salary) with job_id 'MK_MAN', thus, job_id will never be 'MK_MAN'
+
+
 --29. all the information of those employees who did not have any job in the past.
+SELECT *
+FROM employees
+WHERE employee_id NOT IN (
+    SELECT DISTINCT EMPLOYEE_ID
+    FROM job_history
+    );
+
 --30. the employee number, name( first name and last name ) and job title for all employees whose salary is more than any average salary of any department.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name, job_title
+FROM employees
+    INNER JOIN jobs USING (job_id)
+WHERE salary > (
+    SELECT MIN(average_salaries_table.average_salary)
+    FROM (
+         SELECT AVG(salary) AS average_salary
+         FROM EMPLOYEES
+         ) average_salaries_table
+    );
+
 --31. the employee id, name ( first name and last name ) and the job id column with a modified title SALESMAN for those employees whose job title is ST_MAN and DEVELOPER for whose job title is IT_PROG.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name,
+       CASE WHEN job_id = 'ST_MAN' THEN 'SALESMAN'
+           WHEN job_id = 'IT_PROG' THEN 'DEVELOPER'
+           ELSE job_id END AS job_id
+FROM employees;
+
 --32. the employee id, name ( first name and last name ), salary and the SalaryStatus column with a title HIGH and LOW respectively for those employees whose salary is more than and less than the average salary of all employees.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name,
+       CASE WHEN salary >= (SELECT AVG(salary)
+                            FROM employees) THEN 'HIGH'
+            ELSE 'LOW' END AS SalaryStatus
+FROM employees;
+
 --33. the employee id, name ( first name and last name ), SalaryDrawn, AvgCompare (salary - the average salary of all employees)
     -- and the SalaryStatus column with a title HIGH and LOW respectively for those employees whose salary is more than and less than
     -- the average salary of all employees.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name,
+       salary * commission_pct AS SalaryDrawn,
```

Comment:
SalaryDrawn I think is the salary that is stored in the employees table, so we shouldn't use here commission_pct, that is my understanding.

Reply:
Removed **commission_pct**